packages/assets-controllers/src/NftController.ts

```diff
@@ -225,10 +150,14 @@ export interface NftMetadata {
   animation?: string;
   animationOriginal?: string;
   externalLink?: string;
-  creator?: ApiNftCreator;
-  lastSale?: ApiNftLastSale;
+  creator?: string;
   transactionId?: string;
   tokenURI?: string | null;
+  collection?: Collection;
+  address?: string;
+  attributes?: Attributes;
+  lastSale?: LastSale;
+  rarityRank?: string;
```

Comment:
What will happen to existing NFTs already in the controller state?  Will these new fields get populated somehow during a detection?

Reply:
Yes, nftDetectionController detects NFTs every 3mins (on only mainnet for now), so extension NFTs will be pulled again and they get populated again + adding some more data like topBid, lastSale...
If the NFT autodetection is off then users will just manually add the NFT, and it will fetch the data from reservoir